tests/orm/test_querybuilder.py

```diff
@@ -852,6 +852,59 @@ def test_empty_filters(self):
         qb = orm.QueryBuilder().append(orm.Data, filters={'or': [{}, {}]})
         assert qb.count() == count
 
+    from aiida.orm import AbstractCode, Computer, InstalledCode, PortableCode, QueryBuilder
+
+    def test_abstract_code_filtering(self):
+        """Test that querying for AbstractCode correctly returns all code instances.
+
+        This tests the fix for issue #6687, where QueryBuilder couldn't find codes
+        when looking for AbstractCode due to a node_type mismatch.
+        """
+        # Set up test environment
+        computer = orm.Computer(
```

Comment:
Please use an existing fixture [aiida_computer_local](https://aiida.readthedocs.io/projects/aiida-core/en/stable/topics/plugins.html#aiida-computer-local) to simplify this. You can then delete the cleanup code since the fixture does that automatically at the end of the test.